tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/gatherOffset.hlsl

```diff
@@ -35,9 +35,9 @@ float4 main(float4 a : A, float4 b : B) : SV_Target
 {
   uint status;
   float4 r = 0;
-  r += text1.GatherRed(samp1, a.xy, b.xy, b.zw, a.xy, a.zw);
-  r += text1.GatherAlpha(samp1, a.xy, b.xy, b.zw, a.xy, a.zw, status); r += status;
+  r += text1.GatherRed(samp1, a.xy, int2(0, 0), int2(-1, 1), int2(3, 4), int2(-8, 7));
+  r += text1.GatherAlpha(samp1, a.xy, int2(0, 0), int2(-1, 1), int2(3, 4), int2(-8, 7), status); r += status;
 
-  r += text2.GatherBlue(samp1, a.xyz, b.xy, b.zw, a.xy, a.zw, status); r += status;
+  r += text2.GatherBlue(samp1, a.xyz, int2(0, 0), int2(-1, 1), int2(3, 4), int2(-8, 7), status); r += status;
```

Comment:
These old tests flagrantly violating this purported restriction and the lack of any documentation of it do make me wonder if maybe this should be allowed. When next I have access to FXC, I'll see what that does with these.

Reply:
I still think this is correct, but for the record, FXC produces errors on out of range and variable offset parameters for Load and Samples, but not for Gathers.